realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -4118,6 +4128,72 @@ public void minimumDate_runOnMainThreadThrows() {
         }
     }
 
+    private void fillDictionaryTests(){
+        realm.executeTransaction(transactionRealm -> {
+            AllTypes allTypes1 = realm.createObject(AllTypes.class);
+            allTypes1.getColumnStringDictionary().put("hello world1", "Test1");
+
+            AllTypes allTypes2 = realm.createObject(AllTypes.class);
+            allTypes2.getColumnStringDictionary().put("hello world1", "Test2");
+
+            AllTypes allTypes3 = realm.createObject(AllTypes.class);
+            allTypes3.getColumnStringDictionary().put("hello world2", "Test2");
+        });
+    }
+
+    @Test
+    public void dictionary_containsKey(){
+        fillDictionaryTests();
+        RealmResults<AllTypes> results = realm.where(AllTypes.class).containsKey(AllTypes.FIELD_STRING_DICTIONARY, "hello world1").findAll();
+        assertEquals(2, results.size());
+        assertEquals("Test1", results.get(0).getColumnStringDictionary().get("hello world1"));
+        assertEquals("Test2", results.get(1).getColumnStringDictionary().get("hello world1"));
+    }
+
+    @Test
+    public void dictionary_containsKeyNonLatin(){
+        fillDictionaryTests();
+        RealmResults<AllTypes> results = realm.where(AllTypes.class).containsKey(AllTypes.FIELD_STRING_DICTIONARY, "델타").findAll();
+        assertEquals(0, results.size());
+    }
+
+    @Test
+    public void dictionary_containsValue(){
+        fillDictionaryTests();
+        RealmResults<AllTypes> results = realm.where(AllTypes.class).containsValue(AllTypes.FIELD_STRING_DICTIONARY, "Test2").findAll();
+        assertEquals(2, results.size());
+        assertEquals("Test2", results.get(0).getColumnStringDictionary().get("hello world1"));
+        assertEquals("Test2", results.get(1).getColumnStringDictionary().get("hello world2"));
+    }
+
+    @Test
+    public void dictionary_containsEntry(){
+        fillDictionaryTests();
+        RealmResults<AllTypes> results = realm.where(AllTypes.class).containsEntry(AllTypes.FIELD_STRING_DICTIONARY, new AbstractMap.SimpleImmutableEntry<>("hello world1", "Test2")).findAll();
+        assertEquals(1, results.size());
+        assertEquals("Test2", results.first().getColumnStringDictionary().get("hello world1"));
+    }
+
+    @Test
+    public void dictionary_containsKeyNull(){
+        fillDictionaryTests();
+        RealmResults<AllTypes> results = realm.where(AllTypes.class).containsKey(AllTypes.FIELD_STRING_DICTIONARY, null).findAll();
```

Comment:
Just FYI: dictionaries cannot contain null keys. We have a check on the Java side of things and throw an exception. I don't really know how it behaves when passing null keys to the OS, either for queries or writes. It seems it's accepted for queries from what I can see 🤔

Reply:
Queries allow null values anywhere, I am just testing that there is no edge case on dictionaries.